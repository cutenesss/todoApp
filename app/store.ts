import { configureStore, PreloadedState } from '@reduxjs/toolkit'
import RootReducer from './rootReducer'
import Reactotron from './config/ReactotronConfig'
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import { useDispatch, TypedUseSelectorHook, useSelector } from 'react-redux'
import DebugConfig from './config/DebugConfig'

// const enhancers = DebugConfig.reactotron ? [Reactotron.createEnhancer!()] : []

export const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: RootReducer,
    middleware: getDefaultMiddleware => getDefaultMiddleware(),
    preloadedState
  // enhancers,
  })
}

export type RootState = ReturnType<typeof RootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']
export const useAppDispatch: () => AppDispatch = useDispatch
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector
