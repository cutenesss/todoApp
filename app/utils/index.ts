export * from './alertFunction'
export * from './handleSortListTask'
export * from './getPriorityLevel'