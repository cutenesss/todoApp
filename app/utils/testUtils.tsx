import React, { PropsWithChildren } from 'react'
import { render } from '@testing-library/react-native'
import type { RenderOptions } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import type { PreloadedState } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'

import type { RootState } from '../store'
import taskSlice from '../redux/Task/taskSlice'
// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: any
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: { task: taskSlice }, preloadedState }),
  }: ExtendedRenderOptions = {}
) {
  const queries = render(<Provider store={store}>{ui}</Provider>);
  return { ...queries, store };
}