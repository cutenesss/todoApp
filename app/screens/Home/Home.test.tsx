import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react-native';
import HomeScreen from '.';
import { Provider } from 'react-redux';
import {setupStore} from '../../store';
import { renderWithProviders } from '../../utils/testUtils';
import { addTask } from '../../redux';

const testTask = {
  id: 123,
  name: 'name',
  description: 'description'
}

describe("<FloatingButton />", () => {
  const navigateMock = jest.fn();
  const navigation: any = {
    navigate: navigateMock
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={setupStore()}>
          <HomeScreen navigation={navigation} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot();
  });

  it('onPressButton works correctly', () => {
    render(
      <Provider store={setupStore()}>
        <HomeScreen navigation={navigation} />
      </Provider>
    );
    fireEvent.press(screen.getByText("+"));
    expect(navigateMock).toBeCalledTimes(1);
  });
  // it('renders correctly', () => {
  //   const tree = renderer
  //     .create(
  //       <Provider store={store}>
  //         <HomeScreen navigation={navigation} />
  //       </Provider>
  //     )
  //     .toJSON()
  //   expect(tree).toMatchSnapshot();
  // });

  it('onPressItem works correctly', () => {
    const store = setupStore()
    store.dispatch(addTask(testTask))
  
    const { getByTestId } = renderWithProviders(<HomeScreen navigation={navigation} />, { store })
    expect(getByTestId('TaskItem_123')).toBeTruthy();
  });
});