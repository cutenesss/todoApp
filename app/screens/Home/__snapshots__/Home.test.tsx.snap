// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <ViewOverallNumber
        completedTask={0}
        totalTask={0}
      />
    }
    contentContainerStyle={
      Object {
        "paddingBottom": 10,
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
              "paddingHorizontal": 20,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 19,
              }
            }
          >
            Total tasks: 0
          </Text>
          <Text
            style={
              Object {
                "color": "#000",
                "fontSize": 19,
              }
            }
          >
            Completed tasks: 0
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "blue",
        "borderRadius": 40,
        "bottom": 30,
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "position": "absolute",
        "right": 30,
        "width": 40,
        "zIndex": 1,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 25,
            "fontWeight": "bold",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      +
    </Text>
  </View>
</View>
`;
